### ISSUE TITLE
Identifier 'obj' contains type name

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#1823]
Analyzer package: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)
Package Version: 2.6.2-beta2
Diagnostic ID: CA1720

```c#
public static IDisposable ExpectPropertyChanged(this INotifyPropertyChanged obj, int callCount)
{
    // ...
}
```

> CA1720 Identifier 'obj' contains type name

'obj' does not contain a type name in a sense that I understand. Also, I picked `obj` because is a recognized parameter name used by the BCL, e.g. object.Equals. Is this to be avoided? Would something like `observableObject` in fact be better even though it seems less consistent with the BCL?

**Comment by Evangelink** [comment]
@mavasani Is the idea to stop reporting on all `obj` parameters? To allow this name only on some types? Or something else.

**Comment by Evangelink** [comment]
ping

**Comment by Evangelink** [comment]
ping @mavasani

**Comment by sharwell** [comment]
How is `obj` a type name?

**Comment by jnm2** [comment]
I don't get it either.

**Comment by Evangelink** [comment]
Could it be some VB6 legacy or something like that?

**Comment by mavasani** [comment]
Yep, likely a VB specific thing. `Obj` is documented here as special case: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1720?view=vs-2019#rule-description

**Comment by Evangelink** [comment]
@sharwell, @mavasani so how shall we proceed for this ticket? Shall we remove the `obj` check?

**Comment by Evangelink** [comment]
Ping @sharwell and @mavasani

**Comment by sharwell** [comment]
I have no idea. I always just disable this rule so I'm not the best person to ask how it's supposed to work.

**Comment by mavasani** [comment]
@Evangelink It seems reasonable to remove `obj` from the hardcoded list: https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainTypeNames.cs#L31-L70

We should also make this rule configurable by allowing end users to supply additional names that should be flagged by this rule in .editorconfig. You may want to file a separate issue to track that work or combine both changes in one PR.

**Comment by sharwell** [comment]
> We should also make this rule configurable 

I would prefer to do this as part of another rule. As soon as users start trying to special-case their naming rules, it's likely more efficient to use a different set of rules designed for that scenario from the start.

**Comment by mavasani** [comment]
> I would prefer to do this as part of another rule.

Sounds reasonable.
